index.js

```diff
@@ -82,9 +87,108 @@ const sendResponse = ( req, settings, fn ) => {
 		return fn( wpe, null, headers );
 	} );
 
+	if ( processResponseInStreamMode ) {
+		if ( ! req.xhr ) {
+			throw new Error( 'processResponseInStreamMode is not yet implemented for Node.js' );
+		}
+
+		req.xhr.addEventListener( 'readystatechange', event => {
+			if ( event.target.readyState !== XMLHttpRequest.HEADERS_RECEIVED ) {
+				return;
+			}
+
+			const type = event.target.getResponseHeader('Content-Type');
+
+			if ( type.split( ';' )[ 0 ] !== 'application/x-ndjson' ) {
+				return;
+			}
+
+			enableStreamModeProcessing( req, onStreamRecord );
+		} );
+	}
+
 	return req;
 };
 
+// Endpoints in stream mode behave like ordinary endpoints, in that the response contains a JSON
+// representation of some value or WP_Error, but they will also stream other JSON records before
+// that (e.g. progress messages), in application/x-ndjson format.
+//
+// The intent is for the last line of a $stream mode response to be exactly the same as the non-
+// $stream response, but always enveloped as if we were in ?_envelope=1. The other JSON records
+// are also enveloped in the same way, but with .status == 100.
+//
+// I hate enveloping as a matter of principle, but it’s unavoidable in both of these cases. For
+// the last line, which represents the whole response in non-$stream mode, we need to convey the
+// HTTP status code after the body has started. For the other lines, we need an unambiguous way
+// to know that they’re not the last line, so we can exclude it without a “delay line”.
+function enableStreamModeProcessing( req, onStreamRecord ) {
+	if ( ! req.xhr ) {
+		throw new Error( 'processResponseInStreamMode is not yet implemented for Node.js' );
+	}
```

Comment:
You don't need to check this again. It was already checked before calling `enableStreamModeProcessing`

Reply:
fixed in 24f855f48bb41fcd1b4393efbd9d4abcb9b9e963